File: Semester 4/Programming and Projection Enironments/mpp-proiect-java/Proiect_MPP/AirportPersistence/src/main/java/root/persistence/people/PersonRepository.java
package root.persistence.people;

import root.model.people.Person;
import root.persistence.AbstractRepository;

import java.util.Collection;
import java.util.Properties;

public abstract class PersonRepository<T extends Person> extends AbstractRepository<T, Integer> implements IPersonRepository<T> {
    public PersonRepository(Properties properties) {
        super(properties);
        logger.info("Initializing Person Repository ");
    }

    protected abstract String getTableName();

    @Override
    public Collection<T> getPersonByFirstName(String firstName) {
        super.sqlCommand = "SELECT * FROM " + getTableName() + " WHERE first_name='" + firstName + '\'';
        return super.getAll();
    }

    @Override
    public Collection<T> getPersonByLastName(String lastName) {
        super.sqlCommand = "SELECT * FROM " + getTableName() + " WHERE last_name='" + lastName + '\'';
        return super.getAll();
    }
}

File: Semester 4/Programming and Projection Enironments/mpp-proiect-csharp/Proiect_MPP/persistence/people/PersonRepository.cs
using persistence.databaseUtils;
using model.people;
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace persistence.people
{
    public abstract class PersonRepository<T> : AbstractRepository<T, int>, IPersonRepository<T> where T : Person
    {
        public PersonRepository(IDictionary<string, string> properties) : base(properties)
        {
            logger.Info("Initializing Person Repository.");
        }

        protected abstract string getTableName();

        public List<T> getPersonByFirstName(string firstName)
        {
            base.connection = DbUtils.getConnection(base.properties);
            using (base.sqlCommand = connection.CreateCommand())
            {
                base.sqlCommand.CommandText = "SELECT * FROM " + getTableName() + " WHERE first_name=@firstName";

                var paramFirstName = base.sqlCommand.CreateParameter();
                paramFirstName.ParameterName = "@firstName";
                paramFirstName.Value = firstName;
                base.sqlCommand.Parameters.Add(paramFirstName);

                return base.findAll();
            }
        }

        public List<T> getPersonByLastName(string lastName)
        {
            base.connection = DbUtils.getConnection(base.properties);
            using (base.sqlCommand = connection.CreateCommand())
            {
                base.sqlCommand.CommandText = "SELECT * FROM " + getTableName() + " WHERE last_name=@lastName";

                var paramLastName = base.sqlCommand.CreateParameter();
                paramLastName.ParameterName = "@lastName";
                paramLastName.Value = lastName;
                base.sqlCommand.Parameters.Add(paramLastName);

                return base.findAll();
            }
        }
    }
}
